osu.Game/Screens/OnlinePlay/Multiplayer/Match/MatchStartControl.cs

```diff
@@ -101,16 +105,31 @@ protected override void OnRoomLoadRequested()
             endOperation();
         }
 
-        private void onReadyClick()
+        private void onReadyButtonClick()
         {
             if (Room == null)
                 return;
 
             Debug.Assert(clickOperation == null);
             clickOperation = ongoingOperationTracker.BeginOperation();
 
-            if (isReady() && Client.IsHost && !Room.ActiveCountdowns.Any(c => c is MatchStartCountdown))
-                startMatch();
+            if (Client.IsHost)
```

Comment:
This entire `if` block is causing me feelings of unease due to spiraling complexity. One button is now responsible for 3 contextual actions.

I'm not entirely sure what the solution is here, though... I would maybe consider not shoving the abort function into the ready button and instead have it work via 2 separate buttons that are hidden/unhidden depending on room state?